src/openfermion/utils/_trotter_exp_to_qgates.py

```diff
@@ -0,0 +1,277 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+"""Module to perform Trotter-Suzuki decompositions to output as circuits."""
+
+from openfermion.ops import QubitOperator
+import numpy
+import copy
+import collections
+
+"""
+Description:
+    Functions for estimating the exponential of an operator
+    composed of Pauli matrices, by Trotterization. Outputs QASM
+    format to a python stream.
+
+    Change-of-basis gates:
+    H           : Z to X basis (and back)
+    Rx(pi/2)    : Z to Y basis .....
+    Rx(-pi/2)   : Y to Z basis .....
+"""
+
+
+
+def _third_order_trotter_helper(op_list):
+    """Iteratively trotterize the evolution under a QubitOperator according
+        to the scheme: e^(A+B) = e^(7/24 * A) e^(2/3 * B) e^(3/4 * A)
+        e^(-2/3 * B) e^(-1/24 * A) e^(B)
+
+    Note:
+        See N. Hatano and M. Suzuki Lect. Notes Phys 679, 37-68 (2005)
+
+    Args:
+        op_list: (list of QubitOperator) the terms in the
+                hamiltonian as a list of QubitOperators
+    Returns:
+        list of QubitOperators giving the trotterized hamiltonian
+    """
+    # Create list to hold return value
+    ret_val = collections.deque([7.0 / 24.0 * op_list[-2],
+                                2.0 / 3.0 * op_list[-1],
+                                3.0 / 4.0 * op_list[-2],
+                                -2.0 / 3.0 * op_list[-1],
+                                -1.0 / 24.0 * op_list[-2],
+                                1.0 * op_list[-1]])
+
+
+    for i in reversed(range(len(op_list)-2)):
+        temp_ret_val = copy.deepcopy(ret_val)
+        ret_val = collections.deque()
+        ret_val.appendleft(7.0 / 24.0 * op_list[i])
+        ret_val.extend([2.0 / 3.0 * qubop for qubop in temp_ret_val])
+        ret_val.append(3.0 / 4.0 * op_list[i])
+        ret_val.extend([-2.0 / 3.0 * qubop for qubop in temp_ret_val])
+        ret_val.append(-1.0 / 24.0 * op_list[i])
+        ret_val.extend([1.0 * qubop for qubop in temp_ret_val])
+    return ret_val
+
+
+def trotter_operator_grouping(hamiltonian,
+                              trotter_number=1,
+                              trotter_order=1,
+                              term_ordering=None,
+                              k_exp=1.0):
+    """Trotter decomposes operators into groups without exponentiating, to be
```

Comment:
Summary description should be one line, then blank line, then more detailed description